src/main/java/graphql/execution/MergedField.java

```diff
@@ -77,7 +74,7 @@ private MergedField(List<Field> fields) {
      * @return the name of of the merged fields.
      */
     public String getName() {
-        return singleField.getName();
+        return fields.get(0).getName();
```

Comment:
Rather than do a list by index look up everytime we need name, do it once at construction into a `String name` field

Reply:
Or cache the `field` since its used everywhere